pkg/cloudprovider/aws/fleet/packing/packing.go

```diff
@@ -77,34 +79,49 @@ func (p *podPacker) Pack(ctx context.Context, pods []*v1.Pod) ([]*Packing, error
 }
 
 // TODO filter instance types based on node contraints like availability zones etc.
-func (p *podPacker) getNodeCapacities() []*nodeCapacity {
-	return nodeCapacities
+func (p *podPacker) getNodeCapacities(constraints *cloudprovider.Constraints) []*nodeCapacity {
+	p.reserveOnce.Do(func() {
+		for _, nc := range nodeCapacities {
+			kubeletOverhead := binpacking.CalculateKubeletOverhead(nc.total)
+			if ok := nc.reserve(binpacking.MergeResources(constraints.Overhead, kubeletOverhead)); !ok {
+				zap.S().Errorf("Failed to reserve kubelet overhead for node capacity type %v", nc.instanceType)
+			}
+		}
+	})
+	nodeCapacitiesCopy := make([]*nodeCapacity, 0)
+	for _, nc := range nodeCapacities {
+		nodeCapacitiesCopy = append(nodeCapacitiesCopy, &nodeCapacity{nc.instanceType, nc.reserved.DeepCopy(), nc.total.DeepCopy()})
+	}
+	return nodeCapacitiesCopy
+}
+
+type packingsPerCapacity struct {
```

Comment:
nit: joining words like fooPerBar or fooForBar raise naming red flags to me. What do you think about calling this a packingResult? "Result" implies that there's a range of potential outcomes, which makes sense to me w.r.t the fact that pods can be unpacked. 

Alternatively, we could do a diff function (in linear time, so no worse) and only return the packed pods and let the caller sort out the difference.

Reply:
Went with the first approach suggested.